drafts/2020-02-20-Rest-client-generator.adoc

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "The OpenAPI Generator has added support for Eclipse MicroProfile"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/turkeylurkey
+seo-title: OpenAPI Generator has added support for Eclipse MicroProfile - OpenLiberty.io
+seo-description: Develop microservices quicker using the OpenAPI Generator and the Eclipse MicroProfile programming framework.
+blog_description: "Develop microservices quicker using the OpenAPI Generator and the Eclipse MicroProfile programming framework."
+---
+= The OpenAPI Generator has added support for Eclipse MicroProfile
+Paul Gooderham <https://github.com/turkeylurkey>
+
+The OpenAPI Generator can help a developer access a new REST endpoint from their application. It does this by actually generating part of the code required for the application to access that endpoint. The generator can create Java files and also the directory structure corresponding to the package. Recently the OpenAPI Generator was upgraded to support the Microprofile annotations and is an ideal choice for the Open Liberty developer.
+
+If you want to learn more about Microprofile and Open Liberty please check out the educational guide that’s available. You can learn by working on a sample project and you can complete it in less than an hour. https://openliberty.io/guides/microprofile-rest-client.html[Consuming RESTful Java microservices] Pay attention to the sections "Injecting the client with dependency injection" and "Building the client with RestClientBuilder."
+
+== Example
+Say you own a Java web application that you implemented with Microprofile and it is the front end for a collection of stores. Today you are going to add a pet store. You found a pet store back-end service that you're going to use because it provides a RESTful interface. You can use the OpenAPI Generator to accelerate the integration.
```

Comment:
Just to clarify that I understand the scenario correctly... The scenario is to add a pet store to the collection of stores that the app provides a front end to? Is the app something like Amazon.com with different stores behind a single front-end?

Reply:
Can you summarise here that that's the scenario and that the service provides the capability for the developer's app to add pets to the pet store catalog, etc (if that's the case?). I think it needs spelling out so that the next bit about adding a pet flows from this description of the scenario.

Reply:
It already says "it is the front end for a collection of stores." How is that not different stores behind a single front end?

Reply:
Clarified to specify a pet: "The API allows people to list their pets for sale if they provide its name and links to some photos of the pet."